partitionedconsumergroups/README.md

```diff
@@ -0,0 +1,109 @@
+
+## Demo walkthrough
+
+### Static
+
+Create a stream "foo" that automatically partitions over 10 partitions using `static_stream_setup.sh`, then generate some traffic (a new message every 10ms) for that stream using `generate_traffic.sh`.
+
+Create a static consumer group named "cg" on the stream in question, with two members defined called "m1" and "m2": `cg static create balanced foo cg 10 '>' m1 m2`
+
+Start consuming messages with a simulated processing time of 20ms from an instance of member "m1": `cg static consume foo cg m1 --sleep 25ms`. Run in another window cg again to consume as member m2 a second, run multiple instances of members m1 and m2, kill the active one (the one receiving messages) and watch as one of the other instances takes over.
+
+### Elastic
+
+Create a stream 'foo' that captures messages on the subjects `foo.*`, then generate some traffic (a new message every 10ms) for that stream using `generate_traffic.sh`.
+
+Create an elastic consumer group named "cg", partitioning over 10 partitions using the second token (first `*` wildcard in the filter "foo.*") in the subject as the partitioning key: `cg elastic create foo cg 10`.
+
+At this point the elastic consumer group is created, but no members have been added to it yet. But you can start instances of your consuming members already (e.g. `cg elastic consume foo cg m1` for an instance of a member "m1"), for example start instances of members "m1", "m2" and "m3". At this point none of those members are receiving messages. 
+
+Add "m1" and "m2" to the membership: `cg elastic add foo cg m1 m2`, see how they start receiving messages. Then drop "m1" from the membership `cg elastic drop foo cg m1`, add it again, and each time watch as the consumer starts and stops receiving messages, run another consumer "m3" and add/drop it from the membership, etc...
+
+As soon as the elastic consumer group is created, you can start instances of consuming clients (e.g. `cg elastic consume foo cg m1`), and they will start to consume messages as soon as (and as long as) they are in the group's membership.
+
+### Example
+
+To start consuming from a static consumer group, you call `streamconsumergroup.StaticConsume`. To start consuming from an elastic consumer group you call `streamconsumergroup.ElasticConsume`. These calls will return an error and a `ConsumerGroupConsumeContext`. Assuming no error is returned,this will create a Go routine that handles consumption and monitoring for changes in the consumer group's config. 
+
+e.g. for static
+```golang
+consumerGroupContext, err = streamconsumergroup.StaticConsume(myContext, nc, streamName, consumerGroupName, memberName, messageHandler, config)
+```
+The arguments are:
+- `myContext` is a Golang `context.Context` which is going to be used only for the operations that are part of joining the consumer group. You must use `Stop()` on the `ConsumerGroupContext` being returned to stop the consumption.
+- `nc` is a NATS connection object.
```

Comment:
I wonder if we should use `js` as arguments in _all_ methods instead. Using core NATS conn does not allow users to e.g. set API prefix or domain, which I think should be possible. Also, having `js` here would allow setting a default JS timeout to be used in `ElasticConsume()` and `StaticConsume()`

Reply:
Plus it does make sense I think as partitioned consumer groups is an abstraction over jetstream, not core nats. Similar to how `GetBatch()` in `jetstreamext` takes `js` as argument as well.

Reply:
Yes, it was already there as a TODO :). Done.